actions/castVote.ts

```diff
@@ -140,36 +141,87 @@ export async function castVote(
       realm.pubkey,
       proposal.account.governance,
       proposal.pubkey,
-      tokeOwnerRecord.pubkey,
+      tokenOwnerRecord.pubkey,
       governanceAuthority,
       payer,
       undefined,
       message,
       plugin?.voterWeightPk
     )
   }
-  const shouldChunk = votingPlugin?.client instanceof NftVoterClient
-  const instructionsCountThatMustHaveTheirOwnChunk = message ? 4 : 2
-  if (shouldChunk) {
+
+  const ixChunkCount = message ? 4 : 2
+  const isNftVoter = votingPlugin?.client instanceof NftVoterClient
+  const isHeliumVoter = votingPlugin?.client instanceof HeliumVsrClient
+
+  if (!isNftVoter && !isHeliumVoter) {
+    const transaction = new Transaction()
+    transaction.add(...instructions)
+
+    await sendTransaction({ transaction, wallet, connection, signers })
+  }
+
+  // we need to chunk instructions
+  if (isHeliumVoter) {
```

Comment:
Honestly I saw the implementation for castVote for the nftplugin and just copypasta'd it here. I can revisit this if you want. I saw `message ? 4 : 2` for the chunking and Didn't want to dive into it since there were no comments around.

Reply:
its a bit problematic to create generic chunk function some instructions have to be called in one chunk others not + chunking depends on size of single instruction.

Reply:
In case of nfts there is const instructionsCountThatMustHaveTheirOwnChunk = message ? 4 : 2
Because first chunk have to have certain instructions together in one chunk for example if you just cast vote then first chunk is just 2 castVote + withCastPluginVote but if you post msg with your vote we need to have 4 instructions together cast vote + update voter weight + msg + withCastPluginVote then rest i chunked by 2 as i remember.

so in some cases you have chunks where you have [4 inst, 2 inst, 2 inst] in other there is just [2inst, 2inst, 2inst]
 
But i should leave comment there and probably rename the variable because its only chunk of the first instructions that is measured there.